polygeist/lib/Dialect/Polygeist/Transforms/ArgumentPromotion.cpp

```diff
@@ -338,6 +338,19 @@ void Candidate::modifyCallee() {
     polygeist::privatize(funcOp);
 
   LLVM_DEBUG(llvm::dbgs() << "\nNew Callee:\n" << funcOp << "\n";);
+
+  // There could be wrapper function between GPU kernel and kernel body
+  // function. Privatize the wrapper to improve analyses results (e.g., reaching
+  // definition).
+  ModuleOp module = funcOp->getParentOfType<ModuleOp>();
+  SymbolTableCollection symTable;
+  SymbolUserMap userMap(symTable, module);
+  for (Operation *call : userMap.getUsers(funcOp)) {
+    auto caller = call->getParentOfType<FunctionOpInterface>();
+    if (isa<gpu::GPUFuncOp>(caller))
+      continue;
+    polygeist::privatize(caller);
```

Comment:
Should check that the caller has link_once_pdr linkage

Reply:
done.